### PR TITLE
Updated video.js file for variable video ratio

### PR DESCRIPTION
**Pull request by lucasdidthis**
Variable size (ratio) for all video embeds. Testet with a striped down (CSS only) Twenty Nineteen Theme and Wordpress WordPress 5.2.1. A solution for Issue #22.

### CODE REVIEW

File: video.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { onBindFirstLoad } from '../utils/jQueryBindFirst';
```
Comment (kevinweber) [inline video.js]: Why are you adding this new file `video.js` instead of modifying the existing file `modules/shared/video.js`? You shouldn't have to add a new file for this. As of now, this file wouldn't even be used by WordPress, unless you changed something else in your code that is not part of this pull request.

File: video.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      const classes = String(domContainerItem.parentNode.parentNode.classList);
+      const ratioclass = classes.substring(
+        classes.lastIndexOf('wp-embed-aspect-'),
+        classes.lastIndexOf(' '),
+      );
+      const ratioraw = ratioclass.replace('wp-embed-aspect-', '');
+      const splitratio = ratioraw.split('-');
+      newvideoratio = Number(splitratio[1]) / Number(splitratio[0]);
+    }
+    // console.log(newvideoratio);
```
Comment (kevinweber) [inline video.js]: Remove "console.log" so it won't end up in the master branch

File: video.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+export function setBackgroundImage(element, imageUrl) {
+  // Don't simply set "background:url(...)..." because this prop would override
+  // custom styling such as "background-size: cover".
+  element.setAttribute('style', `background-image:url(${imageUrl});background-color:#000;background-position:center center;background-repeat:no-repeat;`);
+}
+
+const debouncedResize = debounce(() => {
+  findElements('.container-lazyload').forEach((domContainerItem) => {
+    /* change ratio */
```
Comment (kevinweber) [inline video.js]: Having to write `/* change ratio */` shouldn't be necessary. This is an indicator that the code needs to be refactored. See my other comments for more concrete suggestions.

File: video.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    const embedelement = domContainerItem;
+    let newvideoratio = 0.5625;
```
Comment (kevinweber) [inline video.js]: Instead of having `const videoratio = 0.5625;` AND `let newvideoratio = 0.5625;`, just have one of those variables. `videoratio` isn't even used anymore, so you should remove it and only use `newvideoratio` (and rename it to `videoratio`).

However, what you really want is not a `let` variable but a function like the following because you won't have to override variables and it'll help you to make the code more readable and reusable:

```
function determineVideoRatio(element) {
    const parent = element && element.parentNode && element.parentNode.parentNode;
    const hasAspectRatioClass = parent && parent.classList.contains('wp-has-aspect-ratio');

    if (hasAspectRatioClass) {
      const classes = String(parent.classList);
      const ratioclass = classes.substring(
        classes.lastIndexOf('wp-embed-aspect-'),
        classes.lastIndexOf(' '),
      );

      const ratioraw = ratioclass.replace('wp-embed-aspect-', '');
      const splitratio = ratioraw.split('-');

      return Number(splitratio[1]) / Number(splitratio[0]);
    }

    return 0.5625; // <-- default
}

...

const videoRatio = determineVideoRatio(embedelement);

...

const height = Math.round(width * videoRatio);
```

Bonus: Now that we have a function, it'll be easier to write tests for this. I'll write those tests once your changes are polished.

File: video.js (hunk 1 above)
Comment (lucasdidthis) [reply video.js]: I did change the existing video.js (in modules/shared). Apperently I did something wrong when uploading it to the git.

File: video.js (hunk 3 above)
Comment (lucasdidthis) [reply video.js]: I inserted the comment for you to easier identify my changes, which isn't really necessary in a git, I suppose.

File: video.js (hunk 4 above)
Comment (lucasdidthis) [reply video.js]: Thanks for the feedback. I'll apply sugested changes as soon as possible. :-)

**Comment by kevinweber** [comment]
Correct. If you actually change the relevant file instead of adding a new
file with copied code, the changes will be obvious.
